packages/beacon-node/src/network/processor/index.ts

```diff
@@ -319,6 +323,9 @@ export class NetworkProcessor {
     executionOptimistic: boolean;
   }): Promise<void> {
     this.isProcessingCurrentSlotBlock = false;
+    if (slot === this.chain.clock.currentSlot) {
+      this.receivedGossipBlock = true;
+    }
```

Comment:
If the network is forked and there are two blocks from different proposers at a slot, both blocks make it through this function, right?  If so does this singular class variable become an issue?

Reply:
> If the network is forked and there are two blocks from different proposers at a slot, both blocks make it through this function, right

no, one if them would fail gossip validation according to the p2p validation

Reply:
> If the network is forked and there are two blocks from different proposers at a slot, both blocks make it through this function, right? If so does this singular class variable become an issue?

yes. But i haven't seen that case in any devnets ever, in that case we'll penalize the peer sending us that block heavily. On the other hand, this fix seems tricky for a small improvement, will revisit it if I find a more legit way to do it